### PR TITLE
Add notes on deprecations and breaking changes for v8.0

### PR DESCRIPTION
**Pull request by amyblais**
Please help fill in any missing information!

### CODE REVIEW

File: source/install/deprecated-features.rst [hunk 1]
```diff
@@ -6,6 +6,12 @@ This page describes features that are removed from support for Mattermost, or wi
 Removed features in upcoming versions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Mattermost Server v8.0.0
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Removing ``ExperimentalSettings.PatchPluginsReactDOM``.
+- Removing deprecated ``PermissionUseSlashCommands``.
+- Removing support for PostgreSQL v10.
```
Comment (agnivade) [inline source/install/deprecated-features.rst:14]: This is still on-hold at the moment.

File: source/install/deprecated-features.rst [hunk 2]
```diff
@@ -6,6 +6,12 @@ This page describes features that are removed from support for Mattermost, or wi
 Removed features in upcoming versions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Mattermost Server v8.0.0
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Removing ``ExperimentalSettings.PatchPluginsReactDOM``.
+- Removing deprecated ``PermissionUseSlashCommands``.
```
Comment (hanzei) [inline source/install/deprecated-features.rst:13]:
```suggestion
- Removing deprecated ``PermissionUseSlashCommands``.
- Removing deprecated ``model.CommandArgs.Session``.
```

File: source/install/self-managed-changelog.md [hunk 3]
```diff
@@ -90,6 +90,16 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
+
+1. Update module version to v8.0. Clients will have to change their module path to the new structure. Very minimal change. There are no API level breaking changes that would require code changes.
+2. Remove ``ExperimentalSettings.PatchPluginsReactDOM``.
+3. Change advanced logging config format. Currently admins have to escape the JSON for advanced logging config in config.json, meaning it must be provided as a single line string.  The breaking change is to allow admins to use multi-line json for ease of composure and reading.  To achieve this the ``model.Config`` struct would have the ``AdvancedLoggingConfig`` fields changed from ``string`` to ``json.RawMessage``.  There are three places this would change: ``LogSettings``,  ``ExperimentalAuditSettings``, and ``NotificationLogSettings``. Existing config.json files with ther escaped string can still be supported;  code would detect the leading quote and treat it as escaped JSON. Unfortunately the field type change will break any code that imports model (plugins, plugin-api, etc).
+4. Remove deprecated ``PermissionUseSlashCommands``.
+5. Remove support for PostgreSQL v10.
```
Comment (hanzei) [inline source/install/self-managed-changelog.md:101]:
```suggestion
4. Remove deprecated ``PermissionUseSlashCommands``.
5. Remove deprecated ``model.CommandArgs.Session``.
6. Remove support for PostgreSQL v10.
```

**Review by hanzei — CHANGES_REQUESTED** [review]
LGTM, but ther

File: source/install/self-managed-changelog.md [hunk 4]
```diff
@@ -90,6 +90,18 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
 
+### Upcoming Deprecations and Breaking Changes in Mattermost v8.0
+
+The following deprecations and breaking changes are planned for the Mattermost v8.0 release, which is scheduled for 2023/06/16. This list is subject to change prior to the release.
+
+1. Update module version to v8.0. Clients will have to change their module path to the new structure. Very minimal change. There are no API level breaking changes that would require code changes.
+2. Remove ``ExperimentalSettings.PatchPluginsReactDOM``.
```
Comment (hmhealey) [inline source/install/self-managed-changelog.md:98]: I'm double checking the first point here, but here's some more information for this change. 
```
2. Remove ``ExperimentalSettings.PatchPluginsReactDOM``. If this setting was previously enabled, confirm that:
    1. All Mattermost-supported plugins are updated to the latest versions.
    2. Any other plugins have been updated to support React 17. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for v7.7 for more information.
```
Also, I don't know if there's a better way to link directly to that note

**Review by mickmister — APPROVED** [review]

**Review by wiersgallak — APPROVED** [review]
Thanks, Amy. Looks good to me.

**Review by hmhealey — APPROVED** [review]

File: source/install/self-managed-changelog.md [hunk 5]
```diff
@@ -90,6 +90,21 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
+2. Remove ``ExperimentalSettings.PatchPluginsReactDOM``. If this setting was previously enabled, confirm that:
+    1. All Mattermost-supported plugins are updated to the latest versions.
+    2. Any other plugins have been updated to support React 17. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for v7.7 for more information.
+3. Change advanced logging config format. Currently admins have to escape the JSON for advanced logging config in config.json, meaning it must be provided as a single line string.  The breaking change is to allow admins to use multi-line json for ease of composure and reading.  To achieve this the ``model.Config`` struct would have the ``AdvancedLoggingConfig`` fields changed from ``string`` to ``json.RawMessage``.  There are three places this would change: ``LogSettings``,  ``ExperimentalAuditSettings``, and ``NotificationLogSettings``. Existing config.json files with ther escaped string can still be supported;  code would detect the leading quote and treat it as escaped JSON. Unfortunately the field type change will break any code that imports model (plugins, plugin-api, etc).
```
Comment (wiggin77) [inline source/install/self-managed-changelog.md:101]:
```suggestion
3. Three configuration fields have been added, `LogSettings.AdvancedLoggingJSON`, `ExperimentalAuditSettings.AdvancedLoggingJSON`, and `NotificationLogSettings.AdvancedLoggingJSON` which support multi-line JSON, escaped JSON as a string, or a filename that points to a file containing JSON.  The `AdvancedLoggingConfig` fields have been deprecated.
```

**Review by cwarnermm — APPROVED** [review]

**Comment by lieut-data** [comment]
@amyblais / @cwarnermm, wondering if I can get your help adding a note about our new `public/` package? Something like the following:

```
The Mattermost public API for Go is now available as a distinctly versioned package. Instead of pinning a particular commit hash, use idiomatic Go to add this package as a dependency:

    go get github.com/mattermost/mattermost-server/server/public

This relocated Go API maintains backwards compatibility with Mattermost v7. Furthermore, the existing Go API previously at github.com/mattermost/mattermost-server/v6/model remains forward compatible with Mattermost v8, but may not contain newer features. Plugins do not need to be recompiled, but developers may opt in to using the new package to simplify their build process.

The new public package is shipping alongside Mattermost v8 as version 0.5.0 to allow for some additional code refactoring before releasing as v1 later this year.
```

**Comment by amyblais** [comment]
@lieut-data Added now via https://github.com/mattermost/docs/pull/6344/commits/ab27245fb83b18baf079ce643d524cdf5aa826cb.

File: source/install/deprecated-features.rst [hunk 6]
```diff
@@ -6,6 +6,16 @@ This page describes features that are removed from support for Mattermost, or wi
...
+Mattermost Server v8.0.0
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Removing ``ExperimentalSettings.PatchPluginsReactDOM``. If this setting was previously enabled, confirm that:
+    1. All Mattermost-supported plugins are updated to the latest versions.
+    2. Any other plugins have been updated to support React 17. See the `Important Upgrade Notes <https://docs.mattermost.com/upgrade/important-upgrade-notes.html>`__ for v7.7 for more information.
+- Removing deprecated ``PermissionUseSlashCommands``.
+- Removing deprecated ``model.CommandArgs.Session``.
+- Removing support for PostgreSQL v10. The new minimum PostgreSQL version will be v11.
+- Deprecating the ``AdvancedLoggingConfig`` fields.
```
Comment (wiggin77) [inline source/install/deprecated-features.rst:18]:
```suggestion
- Deprecating the ``AdvancedLoggingConfig`` fields, and replacing them with ``AdvancedLoggingJSON`` fields which accept inline JSON or a filename.
```

**Pull request merged**